frost-p256/src/lib.rs

```diff
@@ -0,0 +1,277 @@
+            p256::EncodedPoint::from_bytes(buf).map_err(|_| Error::MalformedElement)?;
+        match Option::<AffinePoint>::from(AffinePoint::from_encoded_point(&encoded_point)) {
+            Some(point) => Ok(ProjectivePoint::from(point)),
+            None => Err(Error::MalformedElement),
+        }
+    }
+}
+
+/// Context string from the ciphersuite in the [spec]
+///
+/// [spec]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-04.txt
+
+const CONTEXT_STRING: &str = "FROST-P256-SHA256";
+
+#[derive(Clone, Copy, PartialEq)]
+/// An implementation of the FROST ciphersuite FROST(P-256, SHA-256).
+pub struct P256Sha256;
+
+impl Ciphersuite for P256Sha256 {
+    type Group = P256Group;
+
+    type HashOutput = [u8; 32];
+
+    type SignatureSerialization = [u8; 65];
+
+    /// H1 for FROST(P-256, SHA-256)
+    ///
+    /// [spec]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-04.html#name-frostp-256-sha-256
+    fn H1(m: &[u8]) -> <<Self::Group as Group>::Field as Field>::Scalar {
+        let mut u = [P256ScalarField::zero()];
+        let dst = CONTEXT_STRING.to_owned() + "rho";
+        hash_to_field::<ExpandMsgXmd<Sha256>, Scalar>(&vec![m], dst.as_bytes(), &mut u).unwrap();
+        u[0]
+    }
+
+    /// H2 for FROST(P-256, SHA-256)
+    ///
+    /// [spec]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-04.html#name-frostp-256-sha-256
+    fn H2(m: &[u8]) -> <<Self::Group as Group>::Field as Field>::Scalar {
+        let mut u = [P256ScalarField::zero()];
+        let dst = CONTEXT_STRING.to_owned() + "chal";
+        hash_to_field::<ExpandMsgXmd<Sha256>, Scalar>(&vec![m], dst.as_bytes(), &mut u).unwrap();
+        u[0]
+    }
+
+    /// H3 for FROST(P-256, SHA-256)
+    ///
+    /// [spec]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-04.html#name-frostp-256-sha-256
+    fn H3(m: &[u8]) -> Self::HashOutput {
+        let h = Sha256::new()
+            .chain(CONTEXT_STRING.as_bytes())
+            .chain("digest")
+            .chain(m);
+
+        let mut output = [0u8; 32];
+        output.copy_from_slice(h.finalize().as_slice());
+        output
+    }
+}
+
+// Shorthand alias for the ciphersuite
+type P = P256Sha256;
+
+///
+pub mod keys {
+    use super::*;
+
+    ///
+    pub fn keygen_with_dealer<RNG: RngCore + CryptoRng>(
+        num_signers: u8,
+        threshold: u8,
+        mut rng: RNG,
+    ) -> Result<(Vec<SharePackage>, PublicKeyPackage), &'static str> {
+        frost::keys::keygen_with_dealer(num_signers, threshold, &mut rng)
+    }
+
+    ///
+    pub type SharePackage = frost::keys::SharePackage<P>;
+
+    ///
+    pub type KeyPackage = frost::keys::KeyPackage<P>;
+
+    ///
+    pub type PublicKeyPackage = frost::keys::PublicKeyPackage<P>;
+}
+
+///
+pub mod round1 {
+    use super::*;
+    ///
+    pub type SigningNonces = frost::round1::SigningNonces<P>;
+
+    ///
+    pub type SigningCommitments = frost::round1::SigningCommitments<P>;
+
+    ///
+    pub fn preprocess<RNG>(
```

Comment:
We should probably rename this to match the spec in `frost-core` soon, on at least expose a function `commit()` that matches the spec that calls `preprocess(1, id, rng)`

Reply:
Tracked in https://github.com/ZcashFoundation/frost/issues/78